### PR TITLE
[hotfix] Add layer norm gradients all-reduce for sequence parallel

### PR DESCRIPTION
**Pull request by littsk**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/shardformer/policies/bert.py [hunk 1]
```diff
@@ -148,35 +143,36 @@ def module_policy(self):
...
-        )
+        # optimization configuration
+        if self.shard_config.enable_fused_normalization:
+            # Handle bert layer
+            self.append_or_create_submodule_replacement(
+                description=[
+                    SubModuleReplacementDescription(
+                        suffix="attention.output.LayerNorm",
+                        target_module=col_nn.FusedLayerNorm,
+                        kwargs={"sp_partial_derived": use_sequence_parallel},
```
Comment (littsk) [inline colossalai/shardformer/policies/bert.py:154]: how to solver the case where sequence parallel (sp) is enabled, but fused_normalization is not.

Comment (KKZ20) [reply colossalai/shardformer/policies/bert.py]: Having changed the `norm_cls` value in model_policy()

File: colossalai/shardformer/policies/bert.py [hunk 2]
```diff
@@ -32,6 +32,9 @@
 
 
 class BertPolicy(Policy):
+    def __init__(self) -> None:
+        super().__init__()
```
Comment (FrankLeeeee) [inline colossalai/shardformer/policies/bert.py:36]: Delete this redundant func.

Comment (KKZ20) [reply colossalai/shardformer/policies/bert.py]: Addressed

**Review by FrankLeeeee — APPROVED** [review]

**Review by ver217 — APPROVED** [review]

**Pull request merged**